Project-Aurora/Project-Aurora/Settings/Bindables/BindableNumber.cs

```diff
@@ -0,0 +1,428 @@
+
+                    ushortBindable.Value += Convert.ToUInt16(val);
+                    break;
+
+                case TypeCode.Int16:
+                    var shortBindable = this as BindableNumber<short>;
+                    if (shortBindable == null) throw new ArgumentNullException(nameof(shortBindable), $"Generic type {typeof(T)} does not match actual bindable type {GetType()}.");
+
+                    shortBindable.Value += Convert.ToInt16(val);
+                    break;
+
+                case TypeCode.UInt32:
+                    var uintBindable = this as BindableNumber<uint>;
+                    if (uintBindable == null) throw new ArgumentNullException(nameof(uintBindable), $"Generic type {typeof(T)} does not match actual bindable type {GetType()}.");
+
+                    uintBindable.Value += Convert.ToUInt32(val);
+                    break;
+
+                case TypeCode.Int32:
+                    var intBindable = this as BindableNumber<int>;
+                    if (intBindable == null) throw new ArgumentNullException(nameof(intBindable), $"Generic type {typeof(T)} does not match actual bindable type {GetType()}.");
+
+                    intBindable.Value += Convert.ToInt32(val);
+                    break;
+
+                case TypeCode.UInt64:
+                    var ulongBindable = this as BindableNumber<ulong>;
+                    if (ulongBindable == null) throw new ArgumentNullException(nameof(ulongBindable), $"Generic type {typeof(T)} does not match actual bindable type {GetType()}.");
+
+                    ulongBindable.Value += Convert.ToUInt64(val);
+                    break;
+
+                case TypeCode.Int64:
+                    var longBindable = this as BindableNumber<long>;
+                    if (longBindable == null) throw new ArgumentNullException(nameof(longBindable), $"Generic type {typeof(T)} does not match actual bindable type {GetType()}.");
+
+                    longBindable.Value += Convert.ToInt64(val);
+                    break;
+
+                case TypeCode.Single:
+                    var floatBindable = this as BindableNumber<float>;
+                    if (floatBindable == null) throw new ArgumentNullException(nameof(floatBindable), $"Generic type {typeof(T)} does not match actual bindable type {GetType()}.");
+
+                    floatBindable.Value += Convert.ToSingle(val);
+                    break;
+
+                case TypeCode.Double:
+                    var doubleBindable = this as BindableNumber<double>;
+                    if (doubleBindable == null) throw new ArgumentNullException(nameof(doubleBindable), $"Generic type {typeof(T)} does not match actual bindable type {GetType()}.");
+
+                    doubleBindable.Value += Convert.ToDouble(val);
+                    break;
+            }
+        }
```

Comment:
Is there any reason to use switch trees like this? Why not get the type of `this` and then from there get the generic parameter that has been defined on it and use `Convert.ChangeType` and pass the generic parameter type in? It would mean you could remove a lot of these big chunks of code.

Reply:
I would need to test that out before i can comment further than this is a solution i found from another repo that is MIT licenced.